examples/source_examples/sql_server/connector.py

```diff
@@ -0,0 +1,220 @@
+            last_name NVARCHAR(50) NOT NULL,           -- Employee's last name
+            hire_date DATE NOT NULL,                   -- Date of hire
+            salary INT NOT NULL                        -- Salary
+        );
+        """
+
+    insert_data_sql = """
+        INSERT INTO employee_details (first_name, last_name, hire_date, salary)
+        VALUES 
+            ('John', 'Doe', '2020-05-15', 55000),
+            ('Jane', 'Smith', '2018-03-22', 62000),
+            ('Alice', 'Johnson', '2019-07-30', 58000),
+            ('Bob', 'Brown', '2021-11-01', 54000),
+            ('Charlie', 'Taylor', '2017-06-10', 67000),
+            ('Diana', 'Wilson', '2022-01-20', 51000),
+            ('Eve', 'Martin', '2015-12-15', 75000),
+            ('Frank', 'Moore', '2023-04-05', 52000),
+            ('Grace', 'Hall', '2020-09-14', 60000),
+            ('Hank', 'Lee', '2021-03-18', 53000);
+        """
+
+    try:
+        # Establish connection
+        conn = connect_to_database(configuration)
+        cursor = conn.cursor()
+
+        # Execute the SQL statements
+        print("Creating the table...")
+        cursor.execute(create_table_sql)
+        conn.commit()
+        print("Table created successfully.")
+
+        print("Inserting data into the table...")
+        cursor.execute(insert_data_sql)
+        conn.commit()
+        print("Data inserted successfully.")
+
+    except pyodbc.Error as e:
+        print(f"Database error: {e}")
+
+    finally:
+        # Clean up and close the connection
+        cursor.close()
+        conn.close()
+        print("Connection closed.")
+
+
+
+# Define the update function, which is a required function, and is called by Fivetran during each sync.
+# See the technical reference documentation for more details on the update function
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
+# The state dictionary is empty for the first sync or for any full re-sync
+def update(configuration: dict, state: dict):
+    log.warning("Example: Source Examples - SQL Server")
+
+    # The yield statement returns a generator object.
+    # This generator will yield an upsert operation to the Fivetran connector.
+    # The op.upsert method is called with two arguments:
+    # - The first argument is the name of the table to upsert the data into, in this case, "employee_details".
+    # - The second argument is a dictionary containing the data to be upserted,
+
+    # Connect to your database instance instance.
+    conn = connect_to_database(configuration)
+    cursor = conn.cursor()
+
+    # This is not required. This is just for example illustration purposes.
+    setup_db(configuration)
```

Comment:
If we are creating separate connection for this method, should we move this before line 141?